diff --git a/chrome/browser/chrome_content_browser_client.cc b/chrome/browser/chrome_content_browser_client.cc
index a7d66215ad0b7..bff2929e0a453 100644
--- a/chrome/browser/chrome_content_browser_client.cc
+++ b/chrome/browser/chrome_content_browser_client.cc
@@ -2755,6 +2755,22 @@ void ChromeContentBrowserClient::AppendExtraCommandLineSwitches(
   command_line->CopySwitchesFrom(browser_command_line,
                                  kDinosaurEasterEggSwitches);
 
+  static const char* const kStealthIdentitySwitches[] = {
+      "platform",
+      "locale",
+      "timezone",
+      "screen-width",
+      "screen-height",
+      "screen-avail-width",
+      "screen-avail-height",
+      "screen-avail-left",
+      "screen-avail-top",
+      "screen-color-depth",
+      "device-pixel-ratio",
+      "hardware-concurrency",
+      "device-memory",
+  };
+
 #if BUILDFLAG(IS_CHROMEOS)
   // On Chrome OS need to pass primary user homedir (in multi-profiles session).
   base::FilePath homedir;
@@ -2763,6 +2779,8 @@ void ChromeContentBrowserClient::AppendExtraCommandLineSwitches(
 #endif
 
   if (process_type == switches::kRendererProcess) {
+    command_line->CopySwitchesFrom(browser_command_line,
+                                   kStealthIdentitySwitches);
     content::RenderProcessHost* process =
         content::RenderProcessHost::FromID(child_process_id);
     if (process) {
